doc/release/technotes/README.libraries.rst

```diff
@@ -23,36 +26,141 @@ the ``chpl`` command line take precedence over any found through object
+     ...
+   }
+
+from which one would create a ``.h`` file:
+
+.. code-block:: C
+
+   int64_t bar(void);
+   void baz(int64_t x);
+
+
+Next, update the ``$LD_LIBRARY_PATH`` environment variable to include the
+directory where the new library file lives and the directory where the Chapel
+build lives.  The latter can be found by looking at the output of a
+``$CHPL_HOME/util/printchplenv`` call and finding the appropriate directory
+under ``$CHPL_HOME/lib``; the directory name is of the form
+``$CHPL_HOST_PLATFORM.$CHPL_HOST_COMPILER.(<env variable suffix>-<corresponding environment variable.)+``.
+For instance, if the output of a printchplenv call was::
+
+   CHPL_HOST_PLATFORM: linux64
+   CHPL_HOST_COMPILER: gnu
+   CHPL_TARGET_PLATFORM: linux64
+   CHPL_TARGET_COMPILER: gnu
+   CHPL_TARGET_ARCH: native
+   CHPL_LOCALE_MODEL: flat
+   CHPL_COMM: none
+   CHPL_TASKS: fifo
+   CHPL_LAUNCHER: none
+   CHPL_TIMERS: generic
+   CHPL_MEM: cstdlib
+   CHPL_MAKE: gmake
+   CHPL_ATOMICS: intrinsics
+   CHPL_GMP: gmp
+   CHPL_HWLOC: none
+   CHPL_REGEXP: re2
+   CHPL_WIDE_POINTERS: struct
+   CHPL_LLVM: none
+   CHPL_AUX_FILESYS: none
+
+then the ``$CHPL_HOME/lib/`` subdirectory is::
+
+     linux64.gnu.arch-native.loc-flat.comm-none.tasks-fifo.tmr-generic.mem-cstdlib.atomics-intrinsics.gmp.hwloc-none.re2.wide-struct.fs-none/
+
+.. code-block:: sh
+
+   # Replace $PWD with the appropriate path to your library file if it isn't
+   # in the current directory
+   # Replace libDir with the directory we found from the previous step.
+   export LD_LIBRARY_PATH=$PWD:<libDir>:$LD_LIBRARY_PATH
+
+
+When using a Chapel library from C, one must first initialize the Chapel runtime
+and standard modules.  This is done by the addition of a couple of includes and
+by calling the function ``chpl_library_init()`` before the Chapel library
+function calls and by calling ``chpl_library_finalize()`` after all the Chapel
+library function calls are finished.  For example:
+
+.. code-block:: C
+
+   #include "chpl-string.h"
+   #include "chpl-init.h"
+   #include "testing.h"
+
+   int main(int argc, char* argv[]) {
+       chpl_library_init(argc, argv);
+
+       baz(7); // Call into a library function
+
+       chpl_library_finalize();
 
-      chpl__init_Random();
-      RandomStream* rs = RandomStream_RandomStream(seed);
+       return 0;
+   }
 
-      // ...
+Finally, compilation of the C program involves some additional command line
+includes and links.  The easiest way to get the right combination is from
+the compilation output of another Chapel program.  If one compiles with
+``--print-commands`` and uses all the ``-I`` output and ``-std=c99`` during
+compilation of the C program, one would be most of the way towards a successful
+compile.  All that is missing is the addition of
+``$CHPL_HOME/runtime/src/chpl-init.c`` before the C program file, and
+``-L<libDir> -l<nameOfLib> -L<Chapel Lib location from LD step> -lchpl -lpthread`` in the linking stage.
```

Comment:
The C file needs access to the Chapel headers and libraries if any of the export functions use Chapel types, I'm pretty sure.  When I was experimenting, it kinda escalated from "well, I just need the header for strings" to "but that file requires this file which requires chpl-atomics.h which requires ..." and since I don't think a user would be able to figure out what the next thing to add is as easily, it seemed safer to tell them to use it all than to have them start from nothing and figure it out on their own.